FRAuth/FRAuth/AppIntegrity/FRAppAttestDomainModal.swift

```diff
@@ -0,0 +1,140 @@
+        }
+        guard let bundleIdentifier = bundleIdentifier, !bundleIdentifier.isEmpty else {
+            throw FRDeviceCheckAPIFailure.invalidBundleIdentifier
+        }
+        let userClientData = [challengeKey: challenge,
+                               bundleIdKey: bundleIdentifier]
+        
+        guard let jsonData = try? encoder.encode(userClientData) else {
+            throw FRDeviceCheckAPIFailure.invalidClientData
+        }
+        
+        if !service.isSupported() {
+            throw FRDeviceCheckAPIFailure.featureUnsupported
+        }
+        
+        do {
+            let keyIdentifier = try await service.generateKey()
+            let attest = try await service.attest(keyIdentifier: keyIdentifier, clientDataHash: Data(SHA256.hash(data: challengeUtf8)))
+            let assert = try await service.generateAssertion(keyIdentifier: keyIdentifier, clientDataHash: Data(SHA256.hash(data: jsonData)))
+            return FRAppIntegrityKeys(attestKey: attest.base64EncodedString(),
+                                      assertKey: assert.base64EncodedString(),
+                                      keyIdentifier: keyIdentifier,
+                                      clientDataHash: jsonData.base64EncodedString())
+        }
+        catch let error as DCError {
+            throw FRDeviceCheckAPIFailure.error(code: error.errorCode)
+        }
+        catch {
+            throw error
+        }
+    }
+    
+}
+
+/// Results of AppIntegrity Failures
+/// We need to return some of failures for iOS12, iOS13 devices as well.
+public enum FRDeviceCheckAPIFailure: String, Error {
+    case unknownSystemFailure
+    case featureUnsupported
+    case invalidInput
+    case invalidKey
+    case serverUnavailable
+    case invalidChallenge
+    case invalidBundleIdentifier
+    case invalidClientData
+    case unknownError
+    
+    static func error(code: Int) -> FRDeviceCheckAPIFailure {
+        switch code {
+        case DCError.unknownSystemFailure.rawValue:
+            return .unknownSystemFailure
+        case DCError.featureUnsupported.rawValue:
+            return .featureUnsupported
+        case DCError.invalidInput.rawValue:
+            return .invalidInput
+        case DCError.invalidKey.rawValue:
+            return .invalidKey
+        case DCError.serverUnavailable.rawValue:
+            return .serverUnavailable
+        default:
+            return .unknownError
+        }
+    }
+}
+
+/// List of clientErrors sent to AM
+public enum FRAppIntegrityClientError: String {
+    case unSupported = "Unsupported"
+    case clientDeviceErrors = "ClientDeviceErrors"
+    
+    static func mapError(error: FRDeviceCheckAPIFailure) -> String {
+        switch error {
+         case .featureUnsupported:
+            return FRAppIntegrityClientError.unSupported.rawValue
+        default:
+            return FRAppIntegrityClientError.clientDeviceErrors.rawValue
+        }
+    }
```

Comment:
Instead of this map Error method, I suggest adding a property in FRDeviceCheckAPIFailure like this code below. I think it's cleaner especially when using it 

var clientError: String {
        switch self {
        case .featureUnsupported:
            return FRAppIntegrityClientError.unSupported.rawValue
        default:
            return FRAppIntegrityClientError.clientDeviceErrors.rawValue
        }
    }

Reply:
perfect, i like this one. lets do this.